lib/apress/moysklad/readers/assortment.rb

```diff
@@ -15,18 +15,66 @@ module Readers
       #                                                        password: 'password'
       #                                                      ).each_row { |row| puts row }
       class Assortment < BaseReader
+        class << self
+          def allowed_options
+            super + %i[categories]
+          end
+        end
+
+        def initialize(options)
+          @categories = options.fetch(:categories, [])
+
+          super
+        end
+
+        def each_row
+          if @categories.empty?
+            each_row_offset_loop(nil) { |row| yield presenter.expose(row) }
+          else
+            @categories.each_slice(CATEGORIES_BATCH) do |categories_slice|
+              each_row_offset_loop(categories_filter(categories_slice)) { |row| yield presenter.expose(row) }
+            end
+          end
+        end
+
         private
 
+        CATEGORIES_BATCH = 60
+
         def presenter
           @presenter ||= Presenters::Assortment.new(client)
         end
 
-        def get_data(offset)
+        def each_row_offset_loop(categories_filter, &block)
+          offset = 0
+
+          loop do
+            data = with_retry do
+              get_data(offset, categories_filter)
+            end
+
+            check_rows_size! data, offset
+            data[:rows].each { |row| yield row }
```

Comment:
[Correctable] Style/ExplicitBlockArgument: Consider using explicit block argument in the surrounding method's signature over yield.